input/otlp/traces.go

```diff
@@ -270,7 +270,19 @@ func TranslateTransaction(
 		k := replaceDots(kDots)
 		switch v.Type() {
 		case pcommon.ValueTypeSlice:
-			setLabel(k, event, ifaceAttributeValue(v))
+			switch kDots {
+			case "elastic.profiler_stack_trace_ids":
+				var vSlice = v.Slice()
+				event.Transaction.ProfilerStackTraceIds = make([]string, 0, vSlice.Len())
```

Comment:
We can potentially save an alloc here by:
1. Using https://pkg.go.dev/slices#Grow (requires go 1.21) or https://pkg.go.dev/golang.org/x/exp/slices (requires a new dep). Since apm-server is on 1.21 I think we should also bump apm-data to 1.21
2. Having a nil check and `make` if nil. Otherwise, let it grow on append, if cap is insufficient.

If (1) sounds good, I can bump go version in a separate PR.

Reply:
https://github.com/elastic/apm-data/pull/215

Reply:
merge main, import slices, and use `slices.Grow`
```suggestion
				event.Transaction.ProfilerStackTraceIds = slices.Grow(event.Transaction.ProfilerStackTraceIds, vSlice.Len())
```